Makefile

```diff
@@ -0,0 +1,43 @@
+# These variables follow the naming convention from the GNU Make documentation
+# but their defaults correspond to the rest of the code (note that changing
+# libexecdir here wouldn't affect the default path for the channel plugin
+# directory used by the dameon for example).
+#
+# https://www.gnu.org/software/make/manual/html_node/Directory-Variables.html
+prefix     = /usr
+sbindir    = $(prefix)/sbin
+libexecdir = $(prefix)/libexec
+datadir    = $(prefix)/share
+sysconfdir = /etc
+
+all: pkg = github.com/icinga/icinga-notifications/internal
+all:
+	mkdir -p build
+	go build \
+		-o build/ \
+		-ldflags "-X '$(pkg).LibExecDir=$(libexecdir)' -X '$(pkg).SysConfDir=$(sysconfdir)'" \
+		./cmd/icinga-notifications
+	go build -o build/channel/ ./cmd/channel/...
+
+test:
+	go test ./...
+
+install:
```

Comment:
Add a dependency to `all` to ensure that everything is there.

Reply:
If this Makefile used proper individual build targets with dependencies, yes. But with the rest of the Makefile as it currently is, running `make` and then `make install` will run the `go build` commands twice, which would look strange in the packaging where you typically have separate build and install stages.

Reply:
That's a point. In the current state there is also no dependency on the code to detect when a rebuild might become necessary.